helper/impls/helper-impl-110/src/main/java/com/linkedin/avroutil1/compatibility/avro110/Avro110Adapter.java

```diff
@@ -270,10 +282,23 @@ public FieldBuilder newFieldBuilder(String name) {
             //since avro-compiler may not be on the CP, we use pure reflection to deal with the compiler
             Object compiler = specificCompilerCtr.newInstance(first);
 
-            //configure the compiler for broadest compatibility
+            //configure the compiler
+
+            switch (config.getStringRepresentation()) {
+                case CharSequence:
+                    //this is the (only) way avro 1.4/5 generates code
+                    setStringTypeMethod.invoke(compiler, charSequenceStringTypeEnumInstance);
+                    break;
+                case String:
+                    setStringTypeMethod.invoke(compiler, javaLangStringTypeEnumInstance);
+                    break;
+                case Utf8:
+                    setStringTypeMethod.invoke(compiler, utf8TypeEnumInstance);
+                    break;
+                default:
+                    throw new IllegalStateException("unhandled StringRepresentation " + config.getStringRepresentation());
+            }
```

Comment:
If you adopt the above suggestion of adding the specific variables as enum params, then you could replace this whole switch block with:

```
setStringTypeMethod.invoke(compiler, config.getStringRepresentation().getBlah());
```

Seems attractive since this code is copy-pasted in many adapters...

Reply:
wontt work, tthey wold be the wrong type for the method invocaion

Reply:
Why would they be the wrong type? IDK if I communicated my idea correctly... maybe there's a disconnect?